src/ngraph/runtime/gpu/cuda_emitter.cpp

```diff
@@ -92,33 +101,165 @@ size_t runtime::gpu::CUDAEmitter::build_pad(const runtime::gpu::GPURuntimeContex
         }
 
         codegen::CodeWriter writer;
-        writer << "extern \"C\" __global__ void cuda_" << hash << "(" << dtypes[0] << "* in, "
-               << dtypes[1] << "* out)\n";
+        writer << "extern \"C\" __global__ void cuda_" << hash << "(";
+
+        // if the pad value is static, a runtime argument isn't necessary
+        if (pad_value == "")
+        {
+            writer << dtypes[0] << "* val, ";
+        }
+        writer << dtypes[0] << "* in, " << dtypes[1] << "* out)\n";
         writer.block_begin();
         {
             writer << "size_t tid = blockIdx.x * blockDim.x + threadIdx.x; \n";
+
+            // fill kernel
             writer << "if (tid < " << nthreads << ")\n";
```

Comment:
What happens if this conditional or the one on line 132 fails? Does everything do the right thing? Do we have unit tests for this?

Reply:
These two conditionals allow only the first however many threads to perform the kernel. Since there are almost always more threads than needed for kernel execution (because the gpu schedules threads in warps of 32), these conditionals are needed and are present in many CUDA kernels. Threads that fall outside this range may still execute these operations (due to the SIMT arch.) but are guaranteed not change the program state.

Reply:
@csullivan is right, this is used for padding threads to multiply of warps(i.e. 32).